list.go

```diff
@@ -386,6 +386,9 @@ func (db *RoseDB) popInternal(key []byte, isLeft bool) ([]byte, error) {
 	default:
 		logger.Warn("send to discard chan fail")
 	}
+	if tailSeq-headSeq-1 == 0 {
```

Comment:
You mean if list is empty, delete from indexes, right?
If so, you can move this part to line 347
`if size <= 0 {...}`